src/distributions/range.rs

```diff
@@ -454,9 +469,13 @@ macro_rules! range_float_impl {
                 // Generate a value in the range [1, 2)
                 let value1_2 = (rng.$next_u() >> $bits_to_discard)
                                .into_float_with_exponent(0);
-                // Doing multiply before addition allows some architectures to
-                // use a single instruction.
-                value1_2 * self.scale + self.offset
+                value1_2.mul_add(self.scale, self.offset)
```

Comment:
`mul_add` is much slower than a separate multiply and addition, because it tries to minimize the rounding error. LLVM can turn the two instructions into one for the platforms that support it.

Reply:
> mul_add is much slower

Not according to the [documentation](https://doc.rust-lang.org/nightly/std/primitive.f64.html#method.mul_add).

Reply:
Yes, I read that and tried it before. Can you benchmark it otherwise, because I believe that documentation is wrong.

Reply:
Actually, when using `target-cpu=native`, `mul_add` is faster for `f32`:
```
test distr_range_f32            ... bench:       1,163 ns/iter (+/- 17) = 3439 MB/s
test distr_range_f64            ... bench:       3,852 ns/iter (+/- 6) = 2076 MB/s
```